packages/components/avatar/src/avatar.ts

```diff
@@ -10,80 +12,331 @@ export interface UserProfile {
 export interface UserProfileName {
   first: string;
   last: string;
-  title: string;
+  prefix?: string;
 }
 export interface UserProfilePicture {
   thumbnail: string;
 }
 
+export interface AvatarImage {
+  containerSize: number;
+  size: number;
+  radius: number;
+  y: number;
+  x: number;
+  focusRingPosition: number;
+}
+
+export interface AvatarBadge {
+  height: number;
+  width: number;
+  radius: number;
+  badgeY: number;
+  badgeX: number;
+  badgeBaseX: number;
+  textY?: number;
+  textX?: number;
+}
+
+export interface AvatarIcon {
+  size: number;
+  y?: number;
+  x?: number;
+}
+
 export type AvatarSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl';
 export type AvatarFallbackType = 'initials' | 'image';
 export type AvatarOrientation = 'horizontal' | 'vertical';
 export type UserStatus = 'online' | 'offline' | 'away' | 'do-not-disturb';
 
+let nextUniqueId = 0;
+
+/**
+ * An avatar component to show a picture, initials or icon, to provide a quickly recognisable representation of a user.
+ *
+ * ```html
+ *   <sl-avatar user="{
+ *      name: {
+ *        first: 'Lynn',
+ *        last: 'Smith'
+ *      },
+ *      picture: {
+ *        thumbnail: 'http://sanomalearning.design/avatars/lynn.png'
+ *      }
+ *    }"></sl-avatar>
+ * ```
+ *
+ * @cssproperty --max-width: Max width of the container in ;
+ */
 export class Avatar extends LitElement {
   /** @private */
   static override styles: CSSResultGroup = styles;
 
-  @property() user?: UserProfile;
-  @property({ reflect: true }) size?: AvatarSize = 'md';
+  @property({ type: Object }) user?: UserProfile;
+  @property({ reflect: true }) size: AvatarSize = 'md';
   @property() fallback?: AvatarFallbackType = 'initials';
-  @property({ reflect: true }) orientation?: AvatarOrientation = 'horizontal';
-  @property({ reflect: true }) status?: UserStatus;
+  @property({ reflect: true }) orientation: AvatarOrientation = 'horizontal';
   @property({ type: Boolean, reflect: true, attribute: 'image-only' }) imageOnly?: boolean;
+  @property({ reflect: true }) status?: UserStatus;
+  /** used for Aria-label; you can use `{{badgeText}}` in the string to have it replaced by the value set in the badgeText */
+  @property({ reflect: true }) label: string = '';
+  /**
+   * Experimental feature, use with great caution.
+   */
+  @property({ reflect: true, attribute: 'badge-text' }) badgeText?: string;
+
+  #avatarId = nextUniqueId++;
+  private avatarConfig?: AvatarConfig;
+
+  private imageSizes = {
+    sm: 24,
+    md: 32,
+    lg: 40,
+    xl: 52,
+    '2xl': 64,
+    '3xl': 80
+  };
+
+  private focusRingStrokeWidth = 2;
+  private focusRingStrokeOffset = 2;
+
+  private badgeSizes = {
+    sm: 8,
```

Comment:
The sizes will be the same for all the themes (at least for now) and reading those from the tokens will add an extra step to read the values. Because we also use these values to calculate everything; the size and the positions of the elements we can't just use the tokens in the svg; not all svg-properties can be set using css-vars.
I only read the css-vars that differ per theme and that I find already kind of shakey 😁